editor/src/core/shared/github/helpers.ts

```diff
@@ -850,10 +880,17 @@ export async function refreshGithubData(
   // Collect actions which are the results of the various requests,
   // but not those that show which Github operations are running.
   const promises: Array<Promise<Array<EditorAction>>> = []
-  if (githubAuthenticated) {
-    if (githubUserDetails === null) {
-      promises.push(getUserDetailsFromServer())
-    }
+  if (!githubAuthenticated) {
+    promises.push(Promise.resolve([updateGithubData(emptyGithubData())]))
+  } else if (githubUserDetails === null) {
+    const userDetails = await getUserDetailsFromServer()
+    const actions: EditorAction[] =
+      userDetails == null
+        ? resetGithubStateAndDataActions()
+        : updateUserDetailsFromServerActions(userDetails)
+
+    promises.push(promisifyEditorActions(actions))
```

Comment:
this is very much nitpick / conversation, but to me an obscure helper function always means I have to mentally jump there and then figure out what it does, and then "pop the stack" and continue reading the code. the only exception are "Standard Library" helper functions which we use in a lot of places.

in this specific case, I don't think the readability is worse if we just use the builtin way to promisify something:
```suggestion
    promises.push(Promise.resolve(actions))
```

Reply:
oh nice I did not know it existed 👍